src/libutil/posix-source-accessor.cc

```diff
@@ -134,14 +134,18 @@ SourceAccessor::DirEntries PosixSourceAccessor::readDirectory(const CanonPath &
     DirEntries res;
     for (auto & entry : nix::readDirectory(makeAbsPath(path).string())) {
         std::optional<Type> type;
-        switch (entry.type) {
-        case DT_REG: type = Type::tRegular; break;
-    #ifndef _WIN32
-        case DT_LNK: type = Type::tSymlink; break;
-    #endif
-        case DT_DIR: type = Type::tDirectory; break;
+        // cannot exhaustively enumerate because implementation-specific
+        // additional file types are allowed.
```

Comment:
Shouldn't that be handled by `default`? Seems like a bug in `-Wswitch-enum` that it doesn't recognize the presence of a default case.

Reply:
It is an intentional thing I think: I it wants you to not get complacent because you think you know which things fall under "default", and then are caught making a mistake once the enum definition changes.

There is some other warning I think which is still enabled and accepts this case.